File: ios/NitroGoogleCast/Converters/MediaSeekOptions+toGckMediaSeekOptions.swift
import Foundation
import GoogleCast
import NitroModules

/// Converts a generated `MediaSeekOptions` struct into a `GCKMediaSeekOptions`.
///
/// Reverse lives in `GCKMediaSeekOptions+toMediaSeekOptions.swift`. `position` maps to GCK's
/// `interval`, and `infinite` maps to `seekToInfinite`. `position`/`relative`/`infinite` are
/// non-optional in GCK; when absent we leave GCK's `init()` defaults, so the reverse always
/// emits values (the corpus pins the observed defaults). `resumeState` absent maps to GCK
/// `Unchanged` (→ `nil` on reverse).
extension MediaSeekOptions {
  func toGckMediaSeekOptions() -> GCKMediaSeekOptions {
    let options = GCKMediaSeekOptions()
    if let position { options.interval = position }
    if let relative { options.relative = relative }
    if let infinite { options.seekToInfinite = infinite }
    if let resumeState { options.resumeState = resumeState.toGckResumeState() }
    if let customData { options.customData = customData.toGckCustomData() }
    return options
  }
}

File: android/src/main/java/com/margelo/nitro/googlecast/converters/MediaSeekOptions+toGckMediaSeekOptions.kt
package com.margelo.nitro.googlecast.converters

import com.margelo.nitro.googlecast.MediaSeekOptions
import com.margelo.nitro.googlecast.MediaSeekResumeState
import com.google.android.gms.cast.MediaSeekOptions as GckMediaSeekOptions

/**
 * Converts a generated [MediaSeekOptions] struct into a Google Cast [GckMediaSeekOptions].
 *
 * The reverse lives in `GckMediaSeekOptions+toMediaSeekOptions.kt`. `position` is seconds in
 * the struct / milliseconds (`long`) in GCK. `resumeState` is mapped BY VALUE.
 *
 * The struct's `relative` flag has NO GCK Android counterpart (GCK only models an absolute
 * seek position), so it is dropped here and reported as `null` on the reverse — an
 * iOS/Android reconciliation point for the parity pass.
 */
internal fun MediaSeekOptions.toGckMediaSeekOptions(): GckMediaSeekOptions {
  val builder = GckMediaSeekOptions.Builder()
  position?.let { builder.setPosition((it * 1000).toLong()) }
  infinite?.let { builder.setIsSeekToInfinite(it) }
  resumeState?.let {
    val gckState = when (it) {
      MediaSeekResumeState.PLAY -> GckMediaSeekOptions.RESUME_STATE_PLAY
      MediaSeekResumeState.PAUSE -> GckMediaSeekOptions.RESUME_STATE_PAUSE
    }
    builder.setResumeState(gckState)
  }
  customData?.let { builder.setCustomData(it.toJsonObject()) }
  return builder.build()
}
